dev/archery/archery/linking.py

```diff
@@ -61,9 +64,57 @@ def list_dependency_names(self):
             names.append(name)
         return names
 
+    def _just_symbols(self, symbol_info):
+        return [line.split(' ')[-1] for line in symbol_info if line]
```

Comment:
@kou can we do better here?

Reply:
How about parsing `nm` result before calling these filters something like the following?

```python
result = _nm.run(...)
lines = result.stdout.decode('utf-8').splitlines()
symbol_infos = map(lambda line: re.sub(r'^[\da-fA-F ]+', '', line).split(' ', 1), lines)
symbol_infos = self._remove_weak_symbols(symbol_infos)
symbols = self._extract_symbols(symbol_infos)
```